### PR TITLE
Refactor of Docker Development Setup

### PR DESCRIPTION
**Pull request by BerryDaniel**
Refactor of Docker Development Setup

[See README.md](https://github.com/boundlessgeo/exchange/blob/143b37228d1bd47e561abf422ed4391256d13a7b/docker/README.md) for setup / peer review

Details:
- addition of submodules
- docker dev setup only (vagrant removed)
- ability to use localhost
- utilize stretch Debian / Alpine images for faster startup and smaller container size
- Added Volumes for data persistence
- Ability to use docker setup for Jenkins (tests on slaves).
- Added anywhere integration if repo is in the vendor directory
- Fixed GeoNode version for `capabilities` view

### CODE REVIEW

File: docker/exchange/entrypoint.sh [hunk 1]
```diff
@@ -30,8 +30,17 @@ if [[ $TASK != 'worker' ]]; then
   $manage collectstatic --noinput
   $manage loaddata default_users
   $manage loaddata base_resources
-  $manage loaddata /code/docker/exchange/docker_oauth_apps.json
-  $manage runserver 0.0.0.0:8000
+  $manage loaddata /code/docker/exchange/anywhere.json
```
Comment (mrcnc) [inline docker/exchange/entrypoint.sh]: You can remove this line b/c you're loading it on line 37

File: .env [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
-STORYSCAPES_ENABLED=True
-
-LOOM_STYLING_ENABLED=True
+#exchange
+DEV=True
+MEDIA_ROOT=/app/data/media
+STATIC_ROOT=/tmp/static
+SITEURL=http://localhost/
+ES_URL=http://search:9200/
+ES_UNIFIED_SEARCH=True
```
Comment (tetriscode) [inline .env]: I think this should be `ES_SEARCH=true`

Comment (BerryDaniel) [reply .env]: ES_SEARCH=True

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- maploom
+
+Run the following command to clone all repositories
+
+```bash
+git clone --recursive -j8 git://github.com/boundlessgeo/exchange.git
+cd exchange
+```
+
+Run the following command to clone all repositories
```
Comment (mrcnc) [inline docker/README.md]: You can remove this line...looks like a duplicate of line 32

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
+## Development Environment Setup
+
+**Requirements:**
+- Docker Compose 1.12.0+ (`docker-compose version`)
+- Docker API 1.25+ (`docker version`)
+- Git
+
+Note: You will also need to append nginx to your 127.0.0.1 entry in `/etc/localhosts`.
```
Comment (mrcnc) [inline docker/README.md]: Should this be `/etc/hosts` instead?

File: docker/README.md (hunk 3 above)
Comment (BerryDaniel) [reply docker/README.md]: removed

File: docker/README.md (hunk 4 above)
Comment (BerryDaniel) [reply docker/README.md]: adjusted

File: README.rst [hunk 5]
```diff
@@ -1,61 +1,34 @@
...
-
-**AUDITING**
-
-NOTE: All setting are configurable as environment variables
-
-- AUDIT_ENABLED - Boolean (default is True)
-- AUDIT_TO_FILE - Boolean (default is False)
-- AUDIT_LOGFILE_LOCATION - Full path with filename (default is setting directory filename exchange_audit_log.json)
+The development team recommends using **Docker Engine 1.12.0+** (docker-compose) for setting up your development
+environment. See the README in the docker directory for detailed instructions.
```
Comment (mrcnc) [inline README.rst]: Maybe you can add a link to the README here for maximum developer happiness?  😄

File: docker/exchange/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  # todo: remove this when registry is removed
+  until curl -XPUT "registry:8001/catalog/registry/csw"; do
+    >&2 echo "registry is unavailable - sleeping"
+    sleep 5
+  done
+  $manage migrate --noinput
+  $manage collectstatic --noinput
+  $manage loaddata default_users
+  $manage loaddata base_resources
+  $manage loaddata /code/docker/exchange/anywhere.json
```
Comment (mrcnc) [inline docker/exchange/entrypoint.sh]: You can remove this line b/c you are already loading the data on line 37

File: docker/exchange/anywhere.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+[
+  {
```
Comment (mrcnc) [inline docker/exchange/anywhere.json]: You should remove this record b/c it's already created in `/tmp/exchange/docker/exchange/docker_oauth_apps.json`.  Also, this file should only include fixtures for Anywhere

File: docker/postgis/create_db.sh [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+POSTGRES="psql --username ${POSTGRES_USER}"
+POSTGIS="psql --username ${POSTGRES_USER} --dbname exchange_data"
+
+
+$POSTGRES <<EOSQL
+CREATE DATABASE exchange OWNER ${DB_USER};
+CREATE DATABASE exchange_data OWNER ${DB_USER};
+CREATE DATABASE registry R ${DB_USER};
```
Comment (mrcnc) [inline docker/postgis/create_db.sh]: Is this a typo?  s/R/OWNER

File: docker/nginx/nginx.conf [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        client_max_body_size 1000M;
+
+        location / {
+            proxy_pass         http://exchange:8000;
+        }
+
+        location /geoserver {
+            proxy_pass         http://geoserver:8080/geoserver;
+        }
+
```
Comment (mrcnc) [inline docker/nginx/nginx.conf]: Do you need a reverse proxy for registry also?

Comment (BerryDaniel) [reply docker/nginx/nginx.conf]: No

File: docker/exchange/anywhere.json (hunk 7 above)
Comment (BerryDaniel) [reply docker/exchange/anywhere.json]: done

File: README.rst (hunk 5 above)
Comment (BerryDaniel) [reply README.rst]: done

**Pull request merged**